Add connector_office_365_hr_leave to push leaves to o365

guewen opened the pull request:
It contains the same subset of commits than #77, which needs to be merged first.
I separated the new addon to ease review of #77, but I can do one PR if you prefer.
Only the last commit is relevant here.

This addon pushes the events created for HR leaves to office365.

Commit:
Add connector_office_365_hr_leave to push leaves to o365

Commit:
Add leave name in events created from a leave

Tonow-c2c commented on connector_office_365/models/calendar_event.py:
```diff
@@ -142,24 +207,24 @@ def office_365_open(self):
 
+    @api.model_create_multi
+    def create(self, vals_list):
+        records = super(CalendarEvent, self).create(vals_list)
```
```suggestion
        records = super().create(vals_list)
```

no blocking this just python 3

Tonow-c2c commented on connector_office_365/models/res_users.py:
```diff
@@ -163,7 +169,7 @@ def office_365_request(self, method, url, data=None, headers=None):
                                    client_id=client_id,
                                    client_secret=client_secret)
-        if response.status_code not in [200, 204]:
+        if not response.ok:
```
ok is just [200, 204] or it's wider ?

guewen replied:
Read the commit's message https://github.com/onesteinbv/addons-onestein/pull/78/commits/a61b58eba1f3b69d6af0765c45209b5436a3da9f

Yes, it is wider. It should cover all 2xx and 3xx codes.

Tonow-c2c replied:
ok thanks :+1:

Tonow-c2c commented on connector_office_365_hr_leave/views/hr_leave_views.xml:
```diff
@@ -0,0 +1,29 @@
+      <field name="state" position="before">
+        <field name="need_o365_authenticate" invisible="1"/>
+        <field name="need_o365_manual_push" invisible="1"/>
+        <button string="Office 365 authentication"
```
Any translation ?

Tonow-c2c reviewed (approved):
Just code review

guewen commented:
Rebased

vrenaville committed:
[FIX] set as away by default

vrenaville committed:
[FIX] fix typo in variable name

Commit:
Change user for refusing leave

Commit:
requested changes

Commit:
Use better override

Commit:
update tests

gurneyalex committed:
[IMP] error handling

* use a specific exception class (Office365Error) when API call fails
* when deleting a calendar event in Odoo, don't block if the removal
fails on Office because the record was not found. This can happen if the
meeting was already deleted in Office.

tarteo commented:
I'm sorry for this late response. Could you resolve the conflicts? Then I'll merge it right away.

jcoux commented:
@tarteo PR rebased ;)

tarteo commented:
Thanks!
